chimney/src/main/scala/io/scalaland/chimney/internal/macros/dsl/TransformerDefinitionWhiteboxMacros.scala

```diff
@@ -102,9 +102,23 @@ class TransformerDefinitionWhiteboxMacros(val c: whitebox.Context) extends Macro
   ](f: Tree): Tree = {
     val To = weakTypeOf[To]
     val Inst = weakTypeOf[Inst]
+    val (instType, instSymbol) = if (Inst.typeSymbol.isJavaEnum) {
+      val Function(List(ValDef(_, _, lhs: TypeTree, _)), _) = f
+      lhs.original match {
+        // java enum value in Scala 2.13
+        case SingletonTypeTree(Literal(Constant(t: TermSymbol))) => t.typeInSealedParent(Inst) -> t
+        // java enum value in Scala 2.12
+        case SingletonTypeTree(Select(t, n)) if t.isTerm =>
+          Inst.companion.decls.filter(_.name == n).map(s => s.typeInSealedParent(Inst) -> s).head
```

Comment:
Instead of using unsafe `.head`, would be better to handle empty collection case and explicitly abort compilation with proper error message.